### ISSUE TITLE
Use of relative paths in PATH causes issues

### ISSUE THREAD

**Issue by jekriske** [apptainer/singularity#1140]
## Version of Singularity:

2.4

### Expected behavior
From any directory, the `singularity run` command should run in about the same time, on average.
```
time (singularity run shub://jekriske/lolcow:lowfat > /dev/null)

real	0m1.441s
user	0m0.162s
sys	0m0.169s
```

### Actual behavior
There is a delay from an automounted home directory. It exists even when the file has been previously downloaded.
```
time (singularity run shub://jekriske/lolcow:lowfat > /dev/null)

real	0m8.244s
user	0m0.191s
sys	0m0.197s
```

In our environment $HOME is an NFS share and automounted via autofs. Any directory or file query to /home will trigger a lookup in the automount map. For Singularity, this causes a 6-7 second delay when loading because it's being queried 20 times. 

/var/log/messages
```
Nov  6 15:32:00 servername automount[11370]: key "org" not found in map source(s).
Nov  6 15:32:01 servername automount[11370]: key "org.so" not found in map source(s).
Nov  6 15:32:01 servername automount[11370]: key "orgmodule.so" not found in map source(s).
Nov  6 15:32:01 servername automount[11370]: key "org.py" not found in map source(s).
Nov  6 15:32:02 servername automount[11370]: key "org.pyc" not found in map source(s).
Nov  6 15:32:02 servername automount[11370]: key "org" not found in map source(s).
Nov  6 15:32:02 servername automount[11370]: key "org.so" not found in map source(s).
Nov  6 15:32:03 servername automount[11370]: key "orgmodule.so" not found in map source(s).
Nov  6 15:32:03 servername automount[11370]: key "org.py" not found in map source(s).
Nov  6 15:32:03 servername automount[11370]: key "org.pyc" not found in map source(s).
Nov  6 15:32:04 servername automount[11370]: key "org" not found in map source(s).
Nov  6 15:32:04 servername automount[11370]: key "org.so" not found in map source(s).
Nov  6 15:32:04 servername automount[11370]: key "orgmodule.so" not found in map source(s).
Nov  6 15:32:05 servername automount[11370]: key "org.py" not found in map source(s).
Nov  6 15:32:05 servername automount[11370]: key "org.pyc" not found in map source(s).
Nov  6 15:32:05 servername automount[11370]: key "org" not found in map source(s).
Nov  6 15:32:06 servername automount[11370]: key "org.so" not found in map source(s).
Nov  6 15:32:06 servername automount[11370]: key "orgmodule.so" not found in map source(s).
Nov  6 15:32:06 servername automount[11370]: key "org.py" not found in map source(s).
Nov  6 15:32:07 servername automount[11370]: key "org.pyc" not found in map source(s).
Nov  6 15:32:08 servername Singularity: action-suid (U=#####,P=###)> USER=username, IMAGE='jekriske-lolcow-master-lowfat.simg', COMMAND='run'
```
This search in /home is confirmed by strace. `..` is relative to _where I'm executing the script from_, in this case $HOME so `..` is /home:
```
...
[pid  1225] stat("../org", 0x7ffda3351ef0) = -1 ENOENT (No such file or directory)
[pid  1225] open("../org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1225] open("../orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1225] open("../org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid  1225] open("../org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
...
```

This behavior is caused by:
https://github.com/singularityware/singularity/blob/master/libexec/python/shub/main.py#L26
and
https://github.com/singularityware/singularity/blob/master/libexec/python/shub/api.py#L30

Inside sutils.py the delays are on `import subprocess` and `import tarfile` because they both `import copy`. Copy is what's looking for org.python.core which will not be found. It searches common library locations and PATH. When `..` is added to the PATH then it gets searched too, relative to your current working directory.

### Steps to reproduce behavior

`singularity run shub://jekriske/lolcow:lowfat` from a home directory mounted via autofs auto.home

Here's a simpler example:
```
$ strace python -c "import copy"
...
open("/usr/lib64/python2.6/copy.pyc", O_WRONLY|O_CREAT|O_EXCL|O_TRUNC, 0100644) = -1 EEXIST (File exists)
stat("org", 0x7fff59274350)             = -1 ENOENT (No such file or directory)
open("org.so", O_RDONLY)                = -1 ENOENT (No such file or directory)
open("orgmodule.so", O_RDONLY)          = -1 ENOENT (No such file or directory)
open("org.py", O_RDONLY)                = -1 ENOENT (No such file or directory)
open("org.pyc", O_RDONLY)               = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/org", 0x7fff59274350) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/plat-linux2/org", 0x7fff59274350) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/lib-dynload/org", 0x7fff59274350) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/site-packages/org", 0x7fff59274350) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/site-packages/gtk-2.0/org", 0x7fff59274350) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib/python2.6/site-packages/org", 0x7fff59274350) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
close(3)                                = 0
...
```
vs
```
$ strace python -c "import sys; sys.path.append('..'); import copy"
...
open("/usr/lib64/python2.6/copy.pyc", O_WRONLY|O_CREAT|O_EXCL|O_TRUNC, 0100644) = -1 EEXIST (File exists)
stat("org", 0x7ffe043fcfe0)             = -1 ENOENT (No such file or directory)
open("org.so", O_RDONLY)                = -1 ENOENT (No such file or directory)
open("orgmodule.so", O_RDONLY)          = -1 ENOENT (No such file or directory)
open("org.py", O_RDONLY)                = -1 ENOENT (No such file or directory)
open("org.pyc", O_RDONLY)               = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/org", 0x7ffe043fcfe0) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/plat-linux2/org", 0x7ffe043fcfe0) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/plat-linux2/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/lib-dynload/org", 0x7ffe043fcfe0) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/lib-dynload/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/site-packages/org", 0x7ffe043fcfe0) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib64/python2.6/site-packages/gtk-2.0/org", 0x7ffe043fcfe0) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib64/python2.6/site-packages/gtk-2.0/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/lib/python2.6/site-packages/org", 0x7ffe043fcfe0) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/org.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/orgmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/org.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.6/site-packages/org.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)
stat("..", {st_mode=S_IFDIR|0755, st_size=0, ...}) = 0
stat("..", {st_mode=S_IFDIR|0755, st_size=0, ...}) = 0
stat("../org", 0x7ffe043fcfe0)          = -1 ENOENT (No such file or directory)
open("../org.so", O_RDONLY)             = -1 ENOENT (No such file or directory)
open("../orgmodule.so", O_RDONLY)       = -1 ENOENT (No such file or directory)
open("../org.py", O_RDONLY)             = -1 ENOENT (No such file or directory)
open("../org.pyc", O_RDONLY)            = -1 ENOENT (No such file or directory)
close(3)                                = 0
...
```

One solution is to use `__file__` to determine an absolute path to add.

**Comment by vsoch** [comment]
This is a pretty fantastic bug :) I think this should do the trick:

```
sys.path.append(os.path.dirname(os.path.dirname(__file__)))
```
I'll add it to my current PR for testing.

**Comment by vsoch** [comment]
We will need an abspath in there too.

**Comment by jekriske-lilly** [comment]
This fix looks good for 2.4.1 in the release-2.4 branch

**Comment by dtrudg** [comment]
Closing as this is believed fixed with the merged PR. Thanks!
